src/plugins/query_enhancements/opensearch_dashboards.json

```diff
@@ -4,7 +4,7 @@
   "server": true,
   "ui": true,
   "requiredPlugins": ["data", "opensearchDashboardsReact", "opensearchDashboardsUtils", "savedObjects", "uiActions"],
-  "optionalPlugins": ["dataSource", "usageCollection"],
+  "optionalPlugins": ["dataSource", "usageCollection","assistantDashboards"],
```

Comment:
That means an OSD plugin will depend on a third-party plugin. Should we need to provide some way to register the assistant client instead of adding an optional dependency?

Reply:
I don't think we need any dependency on "assistantDashboards", this is just a API call to check if agent exists, right?

Reply:
Yeah, it's an optional plugin. The OSD will still work fine if this plugin is missing. I'm just wondering if it's suitable to add a third-party plugin dependency in the OSD. I can't find any existing examples that do the same thing.

Reply:
Agree with that, OSD core should not rely on 3rd party plugins.

Reply:
I think we can call the API endpoint directly for now, we will move the entire assistant and summary feature to dashboards-assistant plugin in the future.